Packages/vcs/Lib/vcs2vtk.py

```diff
@@ -507,6 +507,10 @@ def genGrid(data1, data2, gm, deep=True, grid=None, geo=None):
         vg.SetPoints(geopts)
     else:
         vg = grid
+    # Add a GlobalIds array to keep track of cell ids throughout the pipeline
+    globalIds = numpy_to_vtk_wrapper(numpy.arange(0, vg.GetNumberOfCells()), deep=True)
```

Comment:
Because the hardware selector renders and gets the cell ids of the geometry which are not usually the same as the cell ids of the dataset.